### PR TITLE
disallow empty field values for HIT

### LINKED ISSUE #13210: A corner case of command-line parameter overwriting
## Reason
I am meeting a strange corner case that can be produced with the following change
```
diff --git a/test/tests/kernels/2d_diffusion/2d_diffusion_test.i b/test/tests/kernels/2d_diffusion/2d_diffusion_test.i
index cdc4fc2e94..36ddce5e21 100644
--- a/test/tests/kernels/2d_diffusion/2d_diffusion_test.i
+++ b/test/tests/kernels/2d_diffusion/2d_diffusion_test.i
@@ -8,7 +8,14 @@
 ###########################################################
 
 [Mesh]
-  file = square.e
+  type = GeneratedMesh
+  dim =
+  xmin = 0
+  xmax = 165
+  ymin = 0
+  ymax = 165
+  nx = 11
+  ny = 11
 []
 
 [Variables]
```
Even I do `../../../moose_test-opt -i 2d_diffusion_test.i Mesh/dim=2`, I am still getting an error like
```
*** ERROR ***
/Users/wangy2/projects/training/mammoth/moose/test/tests/kernels/2d_diffusion/2d_diffusion_test.i:13: invalid character '='
```
I do not know how the number `13` is from.

However, if the change is
```
diff --git a/test/tests/kernels/2d_diffusion/2d_diffusion_test.i b/test/tests/kernels/2d_diffusion/2d_diffusion_test.i
index cdc4fc2e94..d26b9bc4aa 100644
--- a/test/tests/kernels/2d_diffusion/2d_diffusion_test.i
+++ b/test/tests/kernels/2d_diffusion/2d_diffusion_test.i
@@ -8,7 +8,8 @@
 ###########################################################
 
 [Mesh]
-  file = square.e
+  type = GeneratedMesh
+  dim =
 []
 
 [Variables]
```
I do not see the error.

## Design
Need this fixed so that we can use this in our training materials and still have tests to check the syntax for the rest of the inputs.

## Impact
A bug fix?

### PR DESCRIPTION
**Pull request by rwcarlsen**
Previously:

```
[foo]
  bar =
[]
```

was allowed.  Now it is not.  Related error messages have been improved.
Fixes #13210.

### CODE REVIEW

**Comment by YaqiWang** [comment]
Can we have this with a command-line parameter like `foo/bar=1` now?

**Comment by permcody** [comment]
@YaqiWang - You don't need the parameter to exist to provide it on the CLI. In fact I would discourage it since it's invalid syntax. Just leave the parameter out of the input file altogether and override it on the command line.

**Comment by permcody** [comment]
This is definitely the right fix. The "empty value" should never have been allowed anywhere in the file. You found a genuine HIT "bug".

**Comment by rwcarlsen** [comment]
Or just choose some default to set it to inside the input file.

**Comment by YaqiWang** [comment]
I see. But in this case I do want something in the input like
```
[foo]
  bar =
[]
```
in a template input so that trainee can add the missing parameter during an interactive session. I also want to have a test to make sure this input works fine with the missing parameter being set on the command-line.

**Comment by rwcarlsen** [comment]
Then do `foo = PLACEHOLDER`.

**Comment by YaqiWang** [comment]
That works and actually makes easier to find the missing parameter in a large input file. Is empty parameter nicer in this case? Just want to make sure our trainee have good learning experience.

**Comment by rwcarlsen** [comment]
Well - you don't really have the choice since omitting the field/parameter value is invalid HIT syntax.

**Comment by YaqiWang** [comment]
I see. Will just use `bar = PLACEHOLDER`.

**Review by permcody — APPROVED** [review]
